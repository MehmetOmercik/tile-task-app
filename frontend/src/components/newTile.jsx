export const NewTile = () => {
  return (
    <section className="bg-white h-[500px] w-[300px] rounded-xl p-4">
      <h1 className="text-2xl mb-5">Update Tile</h1>
      <form>
        <label>Launch Date</label>
        <input className="w-full bg-gray-200 rounded-md"></input>
        <label>Status</label>
        <input className="w-full bg-gray-200 rounded-md mb-4"></input>
        <button
          type="submit"
          className="bg-slate-400 rounded-lg p-2 text-lg hover:bg-slate-300"
        >
          Update
        </button>
      </form>
    </section>
  );
};
