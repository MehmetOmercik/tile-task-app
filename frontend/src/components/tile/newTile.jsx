import { useDispatch, useSelector } from "react-redux";
import { updateEditTileObject } from "./tileSlice";
import { DropdownComponent } from "../dropdown";
import { putTile, postTile } from "../../utils/http";
import DateTimePicker from "react-datetime-picker";
import "react-datetime-picker/dist/DateTimePicker.css";
import "react-calendar/dist/Calendar.css";
import "react-clock/dist/Clock.css";
import { isoStringDateFunc } from "../../utils/helpers";

export const NewTile = () => {
  const dispatch = useDispatch();
  const { editTile: editTileBoolean, editTileObject } = useSelector(
    (state) => state.tile
  );
  console.log(editTileObject);

  const handleProperty = (e) => {
    const isoStringDate = isoStringDateFunc(e);
    dispatch(
      updateEditTileObject({
        type: "UPDATE_PROPERTY",
        property: "launchDate",
        value: isoStringDate,
      })
    );
  };

  const handleOptionSelect = (selectedOption) => {
    console.log(selectedOption);
    dispatch(
      updateEditTileObject({
        type: "UPDATE_PROPERTY",
        property: "status",
        value: selectedOption,
      })
    );
  };

  const handleTileUpdateSubmisssion = async (e) => {
    e.preventDefault();
    const tilePayload = {
      id: editTileObject.tileID,
      launch_date: editTileObject.launchDate,
      status: editTileObject.status,
    };
    editTileBoolean
      ? await putTile(editTileObject.tileID, tilePayload)
      : await postTile(tilePayload);
    window.location.reload();
  };

  return (
    <section className="bg-white  w-[300px] rounded-xl p-4">
      <h1 className="text-2xl mb-5">
        {editTileBoolean
          ? `Update Tile ${editTileObject.tileID}`
          : "Create Tile"}
      </h1>
      <form onSubmit={(e) => handleTileUpdateSubmisssion(e)}>
        <label>Launch Date</label>
        <br />
        <DateTimePicker
          dateOnly="true"
          format="yy-MM-dd"
          value={new Date(editTileObject.launchDate)}
          onChange={(e) => handleProperty(e)}
        />
        <br />
        <label>Status</label>
        <DropdownComponent
          options={["live", "pending", "archived"]}
          onSelect={handleOptionSelect}
          defaultValue={editTileObject.status}
        />
        <button
          type="submit"
          disabled={!editTileObject.status}
          className="bg-slate-400 rounded-lg p-2 text-lg hover:bg-slate-300 disabled:bg-red-600"
        >
          {editTileBoolean ? "Update" : "Create"}
        </button>
      </form>
    </section>
  );
};
