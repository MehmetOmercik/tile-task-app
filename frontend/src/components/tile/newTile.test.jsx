import { NewTile } from "./newTile";
import { renderWithProviders } from "../../test/test-utils";

describe("New Tile tests", () => {
  test("Snapshot test", () => {
    const newTile = renderWithProviders(<NewTile />);
    expect(newTile).toMatchSnapshot();
  });
});
