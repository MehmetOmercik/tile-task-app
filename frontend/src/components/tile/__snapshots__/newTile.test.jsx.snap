// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`New Tile tests > Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="bg-white  w-[300px] rounded-xl p-4"
      >
        <h1
          class="text-2xl mb-3"
        >
          Create Tile
        </h1>
        <form>
          <label>
            Launch Date
          </label>
          <br />
          <div
            class="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled mb-3"
          >
            <div
              class="react-datetime-picker__wrapper"
            >
              <div
                class="react-datetime-picker__inputGroup"
              >
                <input
                  hidden=""
                  max="275760-09-13T01:00"
                  min="0001-01-01T00:00"
                  name="datetime"
                  step="60"
                  style="visibility: hidden; position: absolute; z-index: -999;"
                  type="datetime-local"
                  value="2023-11-20T00:00"
                />
                <input
                  autocomplete="off"
                  class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                  data-input="true"
                  inputmode="numeric"
                  max="275760"
                  min="1"
                  name="year"
                  placeholder="----"
                  step="1"
                  type="number"
                  value="2023"
                />
                <span
                  class="react-datetime-picker__inputGroup__divider"
                >
                  -
                </span>
                <input
                  autocomplete="off"
                  class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                  data-input="true"
                  inputmode="numeric"
                  max="12"
                  min="1"
                  name="month"
                  placeholder="--"
                  type="number"
                  value="11"
                />
                <span
                  class="react-datetime-picker__inputGroup__divider"
                >
                  -
                </span>
                <input
                  autocomplete="off"
                  class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                  data-input="true"
                  inputmode="numeric"
                  max="30"
                  min="1"
                  name="day"
                  placeholder="--"
                  type="number"
                  value="20"
                />
              </div>
              <button
                class="react-datetime-picker__clear-button react-datetime-picker__button"
                type="button"
              >
                <svg
                  class="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="4"
                    x2="15"
                    y1="4"
                    y2="15"
                  />
                  <line
                    x1="15"
                    x2="4"
                    y1="4"
                    y2="15"
                  />
                </svg>
              </button>
              <button
                class="react-datetime-picker__calendar-button react-datetime-picker__button"
                type="button"
              >
                <svg
                  class="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                  height="19"
                  stroke="black"
                  stroke-width="2"
                  viewBox="0 0 19 19"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="15"
                    width="15"
                    x="2"
                    y="2"
                  />
                  <line
                    x1="6"
                    x2="6"
                    y1="0"
                    y2="4"
                  />
                  <line
                    x1="13"
                    x2="13"
                    y1="0"
                    y2="4"
                  />
                </svg>
              </button>
            </div>
          </div>
          <br />
          <label>
            Status
          </label>
          <div
            class=""
          >
            <select
              class="p-1 rounded-md mb-4"
            >
              <option
                class=""
                value=""
              >
                Select an option
              </option>
              <option
                class=""
                value="live"
              >
                live
              </option>
              <option
                class=""
                value="pending"
              >
                pending
              </option>
              <option
                class=""
                value="archived"
              >
                archived
              </option>
            </select>
          </div>
          <button
            class="bg-slate-400 rounded-lg p-2 text-lg hover:bg-slate-300 disabled:bg-red-600"
            disabled=""
            type="submit"
          >
            Create
          </button>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="bg-white  w-[300px] rounded-xl p-4"
    >
      <h1
        class="text-2xl mb-3"
      >
        Create Tile
      </h1>
      <form>
        <label>
          Launch Date
        </label>
        <br />
        <div
          class="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled mb-3"
        >
          <div
            class="react-datetime-picker__wrapper"
          >
            <div
              class="react-datetime-picker__inputGroup"
            >
              <input
                hidden=""
                max="275760-09-13T01:00"
                min="0001-01-01T00:00"
                name="datetime"
                step="60"
                style="visibility: hidden; position: absolute; z-index: -999;"
                type="datetime-local"
                value="2023-11-20T00:00"
              />
              <input
                autocomplete="off"
                class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
                data-input="true"
                inputmode="numeric"
                max="275760"
                min="1"
                name="year"
                placeholder="----"
                step="1"
                type="number"
                value="2023"
              />
              <span
                class="react-datetime-picker__inputGroup__divider"
              >
                -
              </span>
              <input
                autocomplete="off"
                class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
                data-input="true"
                inputmode="numeric"
                max="12"
                min="1"
                name="month"
                placeholder="--"
                type="number"
                value="11"
              />
              <span
                class="react-datetime-picker__inputGroup__divider"
              >
                -
              </span>
              <input
                autocomplete="off"
                class="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
                data-input="true"
                inputmode="numeric"
                max="30"
                min="1"
                name="day"
                placeholder="--"
                type="number"
                value="20"
              />
            </div>
            <button
              class="react-datetime-picker__clear-button react-datetime-picker__button"
              type="button"
            >
              <svg
                class="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </svg>
            </button>
            <button
              class="react-datetime-picker__calendar-button react-datetime-picker__button"
              type="button"
            >
              <svg
                class="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </svg>
            </button>
          </div>
        </div>
        <br />
        <label>
          Status
        </label>
        <div
          class=""
        >
          <select
            class="p-1 rounded-md mb-4"
          >
            <option
              class=""
              value=""
            >
              Select an option
            </option>
            <option
              class=""
              value="live"
            >
              live
            </option>
            <option
              class=""
              value="pending"
            >
              pending
            </option>
            <option
              class=""
              value="archived"
            >
              archived
            </option>
          </select>
        </div>
        <button
          class="bg-slate-400 rounded-lg p-2 text-lg hover:bg-slate-300 disabled:bg-red-600"
          disabled=""
          type="submit"
        >
          Create
        </button>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
